sdl_android/src/main/java/com/smartdevicelink/api/lockscreen/SDLLockScreenActivity.java

```diff
@@ -0,0 +1,124 @@
+package com.smartdevicelink.api.lockscreen;
+
+import android.app.Activity;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.graphics.Bitmap;
+
+import android.os.Bundle;
+import android.view.Window;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+
+import com.smartdevicelink.R;
+
+public class SDLLockScreenActivity extends Activity {
+
+	private static final String TAG = "SDLLockScreenActivity";
+	public static final String LOCKSCREEN_COLOR_EXTRA = "LOCKSCREEN_COLOR_EXTRA";
+	public static final String LOCKSCREEN_ICON_EXTRA = "LOCKSCREEN_ICON_EXTRA";
+	public static final String LOCKSCREEN_DEVICE_LOGO_EXTRA = "LOCKSCREEN_DEVICE_LOGO_EXTRA";
+	public static final String LOCKSCREEN_DEVICE_LOGO_BITMAP = "LOCKSCREEN_DEVICE_LOGO_BITMAP";
+	public static final String LOCKSCREEN_CUSTOM_VIEW_EXTRA = "LOCKSCREEN_CUSTOM_VIEW_EXTRA";
+	public static final String LOCKSCREEN_DEVICE_LOGO_DOWNLOADED = "LOCKSCREEN_DEVICE_LOGO_DOWNLOADED";
+	public static final String CLOSE_LOCK_SCREEN_ACTION = "CLOSE_LOCK_SCREEN";
+
+	private final BroadcastReceiver lockScreenBroadcastReceiver = new BroadcastReceiver() {
+		@Override
+		public void onReceive(Context context, Intent intent) {
+			if (intent != null){
+				String action = intent.getAction();
+				if (action != null){
+					if (action.equalsIgnoreCase(CLOSE_LOCK_SCREEN_ACTION)){
+						finish();
+					} else if (action.equalsIgnoreCase(LOCKSCREEN_DEVICE_LOGO_DOWNLOADED)){
+						boolean deviceLogoEnabled = intent.getBooleanExtra(LOCKSCREEN_DEVICE_LOGO_EXTRA, true);
+						Bitmap deviceLogo = intent.getParcelableExtra(LOCKSCREEN_DEVICE_LOGO_BITMAP);
+						if (deviceLogoEnabled && deviceLogo != null){
+							setDeviceLogo(deviceLogo);
+						}
+					}
+				}
+			}
+		}
+	};
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		this.requestWindowFeature(Window.FEATURE_NO_TITLE);
+
+		// set any parameters that came from the lock screen manager
+		initializeActivity(getIntent());
```

Comment:
This call to getIntent() will only return the first intent used to start the activity. You must also override the activity's `onNewIntent` method. Add the following block:
```
@Override
protected void onNewIntent(Intent intent) {
    super.onNewIntent(intent);
    setIntent(intent);
}
```